TFUpdater.py

```diff
@@ -105,6 +105,11 @@ def __init__(self, config, network, initial_learning_rate=1.):
     self.optimizer_vars = []  # type: list[tf.Variable]
     self.optimizer_init_vars_op = None  # type: tf.Operation
 
+    # After graph was build: look if it only uses deterministic ops
+    if self.config.is_true('deterministic_train'):
+      if not check_contains_only_det_ops():
+        print("WARNING: Some op used is not deterministic!")
```

Comment:
The `print` should use `file=log.v1` also.
But better, maybe the function should return which ops (`list[tf.Operation]`) are not deterministic, and then you should print that list here.